src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResultSimulationResource.java

```diff
@@ -0,0 +1,149 @@
+        }
+        catch (URISyntaxException e) {
+            log.error("Error while simulating a submission", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .headers(HeaderUtil.createAlert(applicationName, "An error occurred while simulating a submission: " + e.getMessage(), "errorSubmission")).body(null);
+        }
+    }
+
+    /**
+     * This method is used to notify artemis that there is a new programming exercise build result.
+     * This result is only a SIMULATION for the testing of programming exercises without a connection
+     * to the VCS and CI server
+     * This functionality is only for testing purposes (noVersionControlAndContinuousIntegrationAvailable)
+     * @param exerciseId id of the exercise
+     * @return HTTP OK and Result
+     */
+    @PostMapping(Endpoints.RESULTS_SIMULATION)
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Result> createNewProgrammingExerciseResult(@PathVariable Long exerciseId) {
+        log.debug("Received result notify (NEW)");
+        User user = userService.getUserWithGroupsAndAuthorities();
+        Participant participant = user;
+        ProgrammingExercise programmingExercise = programmingExerciseService.findByIdWithEagerStudentParticipationsAndSubmissions(exerciseId);
+        Optional<StudentParticipation> optionalStudentParticipation = participationService.findOneByExerciseAndParticipantAnyState(programmingExercise, participant);
+
+        if (optionalStudentParticipation.isEmpty()) {
+            return forbidden();
+        }
+
+        ProgrammingExerciseStudentParticipation programmingExerciseStudentParticipation = (ProgrammingExerciseStudentParticipation) optionalStudentParticipation.get();
+        Result result = programmingSubmissionResultSimulationService.createResult(programmingExerciseStudentParticipation);
+
+        messagingService.broadcastNewResult((Participation) optionalStudentParticipation.get(), result);
+        log.info("The new result for {} was saved successfully", ((ProgrammingExerciseStudentParticipation) optionalStudentParticipation.get()).getBuildPlanId());
+        try {
+            return ResponseEntity.created(new URI("/api/results" + result.getId())).body(result);
+        }
+        catch (URISyntaxException e) {
+            log.error("Error while simulating a result", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .headers(HeaderUtil.createAlert(applicationName, "An error occurred while simulating a result: " + e.getMessage(), "errorResult")).body(null);
+        }
+    }
+
+    public static final class Endpoints {
+
+        public static final String ROOT = "/api";
+
+        public static final String SUBMISSIONS_SIMULATION = "/submissions/no-vcs-and-ci-available/{exerciseId}";
+
+        public static final String RESULTS_SIMULATION = "/results/no-vcs-and-ci-available/{exerciseId}";
```

Comment:
/exercises/{exerciseId}/results/no-vcs-and-ci-available

Reply:
!

Reply:
i adapted the url